### ISSUE TITLE
QueuedThreadPool interrupts pool threads when stopped with zero timeout

### ISSUE THREAD

**Issue by wilkinsona** [eclipse/jetty.project#6400]
**Jetty version(s)**

9.4.42

**Java version/vendor** `(use: java -version)`

openjdk version "1.8.0_282"

**OS type/version**

macOS 10.15

**Description**

When `QueuedThreadPool` is configured with a stop timeout of zero it skips trying to stop its threads. This leaves them running until they eventually end after 60s when their idle timeout elapses.

**How to reproduce?**

Running the following code will result in the JVM taking ~60 seconds to exit:

```java
import org.eclipse.jetty.util.thread.QueuedThreadPool;

public class QueuedThreadPoolProblem {

	public static void main(String[] args) throws Exception {
		QueuedThreadPool threadPool = new QueuedThreadPool(2);
		threadPool.setStopTimeout(0);
		threadPool.start();
		threadPool.stop();
	}

}
```

Increasing the timeout to 1ms allows the JVM to exit nigh-on immediately:

```java
import org.eclipse.jetty.util.thread.QueuedThreadPool;

public class QueuedThreadPoolProblem {

	public static void main(String[] args) throws Exception {
		QueuedThreadPool threadPool = new QueuedThreadPool(2);
		threadPool.setStopTimeout(1);
		threadPool.start();
		threadPool.stop();
	}

}
```

**Comment by janbartel** [comment]
Looks like it was the changes around here https://github.com/eclipse/jetty.project/blame/jetty-9.4.x/jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java#L237 with commit 0c61ec3 that changed the behaviour.  Previously, even if the `stopTimeout` was `<=0` the threads would be interrupted almost immediately, now threads are only interrupted if the `stopTimeout` is positive. @gregw comments?

**Comment by peterlynch** [comment]
@gregw this is documented as fixed in 9.4.43.v20210629

https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/VERSION.txt#L6

should this issue be Closed then?

**Comment by joakime** [comment]
PR #6409 is merged.  Closing.
